tiatoolbox/models/engine/engine_abc.py

```diff
@@ -1,21 +1,610 @@
+            the list of pretrained models available at this `link
+            <https://tia-toolbox.readthedocs.io/en/latest/pretrained.html>`_
+            By default, the corresponding pretrained weights will also
+            be downloaded. However, you can override with your own set
+            of weights.
+        weights (str or Path):
+            Path to the weight of the corresponding `model`.
+
+            >>> engine = EngineABC(
+            ...    pretrained_model="pretrained-model-name",
+            ...    weights="pretrained-local-weights.pth")
+
+        batch_size (int):
+            Number of images fed into the model each time.
+        num_loader_workers (int):
+            Number of workers to load the data using :class:`torch.utils.data.Dataset`.
+            Please note that they will also perform preprocessing. default = 0
+        num_post_proc_workers (int):
+            Number of workers to postprocess the results of the model. default = 0
+        device (str):
+            Select the device to run the model. Default is "cpu".
+        verbose (bool):
+            Whether to output logging information.
+
+    Attributes:
+        images (str or :obj:`pathlib.Path` or :obj:`numpy.ndarray`):
+            A NHWC image or a path to WSI.
+        patch_mode (str):
+            Whether to treat input image as a patch or WSI.
+            default = True.
+        model (str | nn.Module):
+            Defined PyTorch model.
+            Name of the existing models supported by the TIAToolbox for
+            processing the data. For a full list of pretrained models,
+            refer to the `docs
+            <https://tia-toolbox.readthedocs.io/en/latest/pretrained.html>`_
+            By default, the corresponding pretrained weights will also
+            be downloaded. However, you can override with your own set
+            of weights via the `weights` argument. Argument
+            is case-insensitive.
+        ioconfig (ModelIOConfigABC):
+            Input IO configuration to run the Engine.
+        _ioconfig ():
+            Runtime ioconfig.
+        return_labels (bool):
+            Whether to return the labels with the predictions.
+        merge_predictions (bool):
+            Whether to merge the predictions to form a 2-dimensional
+            map. This is only applicable `patch_mode` is False in inference.
+        resolution (Resolution):
+            Resolution used for reading the image. Please see
+            :obj:`WSIReader` for details.
+        units (Units):
+            Units of resolution used for reading the image. Choose
+            from either `level`, `power` or `mpp`. Please see
+            :obj:`WSIReader` for details.
+        patch_input_shape (tuple):
+            Size of patches input to the model. Patches are at
+            requested read resolution, not with respect to level 0,
+            and must be positive.
```

Comment:
We call this shape in the arg name but then immediately call it size in the description. Consistent language would help.